### PR TITLE
use kratos to replace uc

### PR DESCRIPTION
**Pull request by Effet**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
kratos is good enough for us to replace uc

#### Which issue(s) this PR fixes:

Fixes #

#### Specified Reivewers:
/assgin @sfwn 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation usage docs, etc.:

```docs

```

### CODE REVIEW

File: modules/cmdb/conf/conf.go [hunk 1]
```diff
@@ -101,6 +101,11 @@ type Conf struct {
...
+
+	// ory/kratos config
+	OryEnabled           string `default:"false" env:"ORY_ENABLED"`
```
Comment (sfwn) [inline modules/cmdb/conf/conf.go:106]: use bool is better.

File: modules/cmdb/conf/conf.go [hunk 2]
```diff
@@ -500,3 +505,21 @@ func EnableNS() bool {
...
+func OryKratosAddr() string {
+	return cfg.OryKratosAddr
+}
+
+func OryKratosPrivateAddr() string {
+	if cfg.OryKratosPrivateAddr != "" {
+		return cfg.OryKratosPrivateAddr
+	} else if cfg.OryKratosAddr != "" {
+		// TODO: ugly hack!
+		return cfg.OryKratosAddr[:len(cfg.OryKratosAddr)-1] + "4"
```
Comment (sfwn) [inline modules/cmdb/conf/conf.go:522]: please add comment and examples for this ugly hack.

File: modules/cmdb/initialize.go [hunk 3]
```diff
@@ -211,7 +211,10 @@ func initEndpoints() (*endpoints.Endpoints, error) {
 	}
 
 	// 初始化UC Client
-	uc := utils.NewUCClient()
+	uc := ucauth.NewUCClient(discover.UC(), conf.UCClientID(), conf.UCClientSecret())
+	if conf.OryEnabled() {
+		uc = ucauth.NewUCClient(conf.OryKratosPrivateAddr(), "kratos", "")
```
Comment (sfwn) [inline modules/cmdb/initialize.go:216]: use const instead of magic string.

File: modules/openapi/loginserver.go [hunk 4]
```diff
@@ -245,24 +249,30 @@ func (s *LoginServer) Logout(rw http.ResponseWriter, req *http.Request) {
...
-		logrus.Error(errStr)
-		http.Error(rw, errStr, http.StatusBadGateway)
-		return
-	}
-	reqDomain, err := conf.GetDomain(req.Host, conf.CookieDomain())
-	if err != nil {
-		http.Error(rw, err.Error(), http.StatusUnauthorized)
-		return
+	if conf.OryEnabled() {
+		user := auth.NewUser(s.auth.RedisCli)
```
Comment (sfwn) [inline modules/openapi/loginserver.go:253]: maybe OrgDisabled()

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/cmdb/conf/conf.go (hunk 2 above)
Comment (Effet) [reply modules/cmdb/conf/conf.go]: rollback to simple config, no hack

File: modules/cmdb/services/approve/approve.go [hunk 5]
```diff
@@ -62,7 +61,7 @@ func WithDBClient(db *dao.DBClient) Option {
 }
 
 // WithUCClient 配置 uc client
```
Comment (luoyuctl) [inline modules/cmdb/services/approve/approve.go:63]: Do not use chinglish

File: modules/cmdb/services/audit/audit.go [hunk 6]
```diff
@@ -17,26 +17,24 @@ import (
...
 	"github.com/erda-project/erda/modules/cmdb/dao"
 	"github.com/erda-project/erda/modules/cmdb/model"
-	"github.com/erda-project/erda/modules/cmdb/utils"
 	"github.com/erda-project/erda/pkg/cron"
 	"github.com/erda-project/erda/pkg/excel"
+	"github.com/erda-project/erda/pkg/ucauth"
+	"github.com/sirupsen/logrus"
 )
 
 // Audit 成员操作封装
```
Comment (luoyuctl) [inline modules/cmdb/services/audit/audit.go:34]: Do not use chinglish

File: modules/cmdb/services/approve/approve.go (hunk 5 above)
Comment (luoyuctl) [inline modules/cmdb/services/approve/approve.go:63]: Do not use chinglish

File: modules/cmdb/services/audit/audit.go [hunk 7]
```diff
@@ -144,7 +142,7 @@ func (a *Audit) convertAuditsToExcelList(audits []model.Audit) ([][]string, erro
 		return nil, err
 	}
 	for _, u := range users {
-		userIDNameMap[strconv.FormatUint(u.ID, 10)] = u.Nick
+		userIDNameMap[u.ID] = u.Nick
 	}
 
 	// 把r里的userID替换成userName
```
Comment (luoyuctl) [inline modules/cmdb/services/audit/audit.go:148]: Bad annotation.

File: modules/cmdb/services/audit/audit.go [hunk 8]
```diff
@@ -64,7 +62,7 @@ func WithDBClient(db *dao.DBClient) Option {
 }
 
 // WithUCClient 配置 uc client
```
Comment (luoyuctl) [inline modules/cmdb/services/audit/audit.go:64]: Bad annotation.

**Comment by luoyuctl** [comment]
In my opinion,do not use as much as possible chinglish, it will make reading comprehension more difficult.

File: modules/cmdb/services/audit/audit.go (hunk 7 above)
Comment (Effet) [reply modules/cmdb/services/audit/audit.go:148]: that's old stuff, leave here, we have a plan to refactor all of codes. until then, all chinese text will be deleted.

File: modules/cmdb/services/audit/audit.go (hunk 8 above)
Comment (Effet) [reply modules/cmdb/services/audit/audit.go:64]: that's old stuff, leave it here, we have a plan to refactor all the code. until then, all chinese text will be deleted.

File: modules/cmdb/services/approve/approve.go (hunk 5 above)
Comment (Effet) [reply modules/cmdb/services/approve/approve.go:63]: I guess what you mean is that there should be no chinese text. and,

that's old stuff, leave it here, we have a plan to refactor all the code. until then, all chinese text will be deleted.

File: modules/cmdb/services/audit/audit.go (hunk 6 above)
Comment (Effet) [reply modules/cmdb/services/audit/audit.go:34]: I guess what you mean is that there should be no chinese text. and,

that's old stuff, leave it here, we have a plan to refactor all the code. until then, all chinese text will be deleted.

**Comment by Effet** [comment]
> In my opinion,do not use as much as possible chinglish, it will make reading comprehension more difficult.

I totally aggress with that. and we have a plan to refactor all the code. until then, all chinese text will be deleted.

**Review by sfwn — APPROVED** [review]

**Pull request merged**